python/cudf_polars/cudf_polars/dsl/expr.py

```diff
@@ -1157,6 +1170,22 @@ def _last(self, column: Column) -> Column:
         n = column.obj.size()
         return Column(plc.copying.slice(column.obj, [n - 1, n])[0])
 
+    def _quantile(
+        self, column: Column, quantile: Column | pa.Scalar, *, interp: str
+    ) -> Column:
+        # do_evaluate should have evaluated our quantile Column to a pyarrow scalar
+        if not isinstance(quantile, pa.Scalar):
+            raise ValueError(  # noqa: TRY004
+                "cudf-polars only supports expressions that evaluate to a scalar as the quantile argument"
+            )
```

Comment:
Since we only handle the case of `quantile` being a `Literal`, please raise if that is not observed in `__init__` and then you don't need this handling here. We can just write:

```
def _quantile(self, column: Column, quantile: pa.Scalar, *, interp: str) -> Column:
    return self._reduce(column, request=...)
```